packages/client/lib/client/index.spec.ts

```diff
@@ -103,6 +104,21 @@ describe('Client', () => {
       },
       minimumDockerVersion: [6, 2]
     });
+
+    testUtils.testWithClient('should accept a credentialSupplier', async client => {
+      assert.equal(
+          await client.ping(),
+          'PONG'
+      );
+    }, {
+      ...GLOBAL.SERVERS.PASSWORD,
+      clientOptions: {
+        // simulate a slight pause to fetch the credentials
```

Comment:
Does "simulating a pause" make the test better in any way? can't we just `Promise.resolve` directly?

Reply:
I found that the order of the resolution of promises could cause a problem if it was just an instantly resolving promise, that edge case wouldn't be covered.